### PR TITLE
Add example Bullmq processor

### PR DESCRIPTION
**Pull request by kibertoad**

### CODE REVIEW

File: src/infrastructure/jobs/AbstractInMemoryJob.ts [hunk 1]
```diff
@@ -23,11 +24,29 @@ export type LockConfiguration = {
...
+          name: error.name,
+          message: error.message,
+          stack: error.stack,
+        }),
+      )
+    },
+  )
+}
+
+export abstract class AbstractInMemoryJob {
```
Comment (CarlosGamero) [inline src/infrastructure/jobs/AbstractInMemoryJob.ts:45]: I am not the best with names definitely, but wondering why not call it is just AbstractCronJob, I mean, at the end this is a cron job

Comment (kibertoad) [reply src/infrastructure/jobs/AbstractInMemoryJob.ts:45]: Makes sense. It is not always Cron, though, you can also define a human-readable interval, e. g. "every 30 minutes".

What do you think about AbstractPeriodicJob or AbstractRecurringJob?

File: src/infrastructure/jobs/AbstractRedisJobProcessor.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+} from '@lokalise/background-jobs-common'
+import type { RequestContext } from '@lokalise/fastify-extras'
+import type { Job } from 'bullmq'
+
+import { isTest } from '../config'
+import type { Dependencies } from '../diConfig'
+
+export type BackgroundJobConfig = Omit<BackgroundJobProcessorConfig, 'isTest'>
+
+export abstract class AbstractRedisJobProcessor<
```
Comment (CarlosGamero) [inline src/infrastructure/jobs/AbstractRedisJobProcessor.ts:14]: If we decide to rename the `InMemory` job, I guess we can just rename this as `AbstractJobProcessor` or something like that but no strong opinion here

File: src/infrastructure/jobs/AbstractInMemoryJob.ts (hunk 1 above)
Comment (kibertoad) [reply src/infrastructure/jobs/AbstractInMemoryJob.ts:45]: AbstractPeriodicJob it is, then!

File: src/infrastructure/jobs/AbstractRedisJobProcessor.ts (hunk 2 above)
Comment (CarlosGamero) [reply src/infrastructure/jobs/AbstractRedisJobProcessor.ts:14]: yeah, you are rigjht ... 😓 not sure which is a good pattern here sorry, if yo think that keeping Redis in the name is fine, I am ok with that

Comment (VSevostianov) [reply src/infrastructure/jobs/AbstractRedisJobProcessor.ts:14]: ```AbstractEnqueuedJobProcessor```? Not the best with naming today:)

Comment (kibertoad) [reply src/infrastructure/jobs/AbstractRedisJobProcessor.ts:14]: I'm scared of what you can do when you are at your best then, mortals are not supposed to wield that much power.

This is perfect, thanks!

**Review by CarlosGamero — APPROVED** [review]

**Pull request merged**